snabble/Android-SDK
# Convert events.kt to kotlin and fix state bug in the paymentstatusview

### What's new?

Events.kt has been cleaned up and migrated to kotlin. Also a new observer is added to the paymentStatusView, which observes the `currentCheckedInShop`. This is need to avoid state bugs, appearing due to null values or different shopId's.

### How to test?

* the `paymentStatusView` can be accessed without being checked in
  * Set the property `Snabble.currentCheckedInShop` to null, when the user puts the app into the background.
  * Launch the application, and buy something, then move the app into the background.
  * Reopen the application and select the scanner. -> No status view 
  * Checkin into the shop and switch to the scanner. -> No status view should appear since it was finished on the try bevore

* the `paymentStatusView` can be accessed with the wrong shop checked in
  * Set the property `Snabble.currentCheckedInShop` to null, when the user puts the app into the background.
  * Launch the application, and buy something, then move the app into the background. 
  * Checkin into a different shop and switch to the scanner. -> No status view should appear since the shop id is different

In both cases the chechoutState should be in its initial State NONE after the process.
